server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsAction.java

```diff
@@ -64,9 +100,93 @@ public String getName() {
+
+        if (!metrics.isEmpty()) {
+            if (metrics.size() > 1 && metrics.contains("_all")) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        Locale.ROOT,
+                        "request [%s] contains _all and individual metrics [%s]",
+                        request.path(),
+                        request.param("metric")
+                    )
+                );
+            }
+
+            if (indexMetrics.size() > 1 && indexMetrics.contains("_all")) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        Locale.ROOT,
+                        "request [%s] contains _all and individual index metrics [%s]",
+                        request.path(),
+                        request.param("index_metric")
+                    )
+                );
+            }
+
+            clusterStatsRequest.applyMetricFiltering(true);
+
+            final Set<String> metricsRequested = new HashSet<>();
+            if (metrics.contains("_all")) {
+                metricsRequested.addAll(METRIC_REQUEST_CONSUMER_MAP.keySet());
+            } else {
+                metricsRequested.addAll(metrics);
+            }
+            final Set<String> invalidMetrics = new TreeSet<>();
+            for (String metric : metricsRequested) {
+                Consumer<ClusterStatsRequest> clusterStatsRequestConsumer = METRIC_REQUEST_CONSUMER_MAP.get(metric);
+                if (clusterStatsRequestConsumer != null) {
+                    clusterStatsRequestConsumer.accept(clusterStatsRequest);
+                } else {
+                    invalidMetrics.add(metric);
+                }
+            }
+            if (!invalidMetrics.isEmpty()) {
+                throw new IllegalArgumentException(unrecognized(request, invalidMetrics, METRIC_REQUEST_CONSUMER_MAP.keySet(), "metric"));
+            }
+            if (!metricsRequested.contains(ClusterStatsRequest.Metric.INDICES.metricName()) && !indexMetrics.isEmpty()) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        Locale.ROOT,
+                        "request [%s] contains index metrics [%s] but indices stats not requested",
+                        request.path(),
+                        request.param("index_metric")
+                    )
+                );
+            }
+
+            if (metricsRequested.contains(ClusterStatsRequest.Metric.INDICES.metricName())) {
+                final Set<String> indexMetricsRequested = new HashSet<>();
+                if (indexMetrics.isEmpty() || indexMetrics.contains("_all")) {
+                    indexMetricsRequested.addAll(INDEX_METRIC_TO_REQUEST_CONSUMER_MAP.keySet());
+                } else {
+                    indexMetricsRequested.addAll(indexMetrics);
+                }
+                final Set<String> invalidIndexMetrics = new TreeSet<>();
```

Comment:
validation required only if `indexMetricsRequested` is not empty

Reply:
This will introducing more branching in the code , making it difficult to understand the filtering logic.

Reply:
If the default values are set by the code and not coming from request, it is un-necessary to validate them. Would an early branching help ?